src/main/scala/dicom/json/DicomJson.scala

```diff
@@ -51,11 +53,70 @@ object DicomJson {
 
   def json2string(json: JsonStructure): String = {
     val sw = new java.io.StringWriter
-    val jwf =
-      javax.json.Json.createWriterFactory(Map.empty.asJava)
+    val jwf = Json.createWriterFactory(Map.empty.asJava)
     val jw = jwf.createWriter(sw)
     jw.write(json)
     jw.close
     sw.toString
   }
+
+  def deepRenameJsonKeys(
+      jsonArr: JsonArray,
+      renameF: String => String
+  ): JsonArray = {
+    val outBuilder = Json.createArrayBuilder
+    jsonArr.forEach(jsonVal => {
+      val outVal = jsonVal.getValueType match {
+        case JsonValue.ValueType.OBJECT =>
+          deepRenameJsonKeys(jsonVal.asJsonObject, renameF)
+        case JsonValue.ValueType.ARRAY =>
+          deepRenameJsonKeys(jsonVal.asJsonArray, renameF)
+        case _ => jsonVal
+      }
+      outBuilder.add(outVal)
+    })
+    outBuilder.build
+  }
+
+  def deepRenameJsonKeys(
+      jsonObj: JsonObject,
+      renameF: String => String
+  ): JsonObject = {
+    val outBuilder = javax.json.Json.createObjectBuilder
+    jsonObj.entrySet.asScala.foreach(entry => {
+      val outKey = renameF(entry.getKey)
+      val outValue = entry.getValue.getValueType match {
+        case JsonValue.ValueType.OBJECT =>
+          deepRenameJsonKeys(entry.getValue.asJsonObject, renameF)
+        case JsonValue.ValueType.ARRAY =>
+          deepRenameJsonKeys(entry.getValue.asJsonArray, renameF)
+        case _ => entry.getValue
+      }
+      outBuilder.add(outKey, outValue)
+    })
+    outBuilder.build
+  }
+
+  def deepRenameJsonKeyTagToKeyword(
+      jsonStruct: JsonStructure
+  ): JsonStructure = {
+    def renameTagToKeywordOrFallback(key: String) =
+      scala.util
+        .Try(Integer.parseInt(key, 16))
+        .map(intTag =>
+          DicomStandardDictionary.tagMap
+            .get(intTag)
+            .map(elemDict => elemDict.keyword)
+        )
+        .toOption
+        .flatten
+        .getOrElse(key)
```

Comment:
I'd move the function out of the scope of the outer function since it has no dependency from its scope